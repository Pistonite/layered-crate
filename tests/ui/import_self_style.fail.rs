#![allow(unused)]

mod current {
    pub struct MyTypeInThisLayer;
    pub struct MyDepInThisLayer;
}

mod test {
    #[layered_crate::import]
    use current::{
        MyTypeInThisLayer,
        self::MyDepInThisLayer,
    };

}

mod test2 {
    // this should not produce any error
    #[layered_crate::import]
    use current::{
        self,
        {MyTypeInThisLayer, MyDepInThisLayer},
    };
}


fn main() {}
