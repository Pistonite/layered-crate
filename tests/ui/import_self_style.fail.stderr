error: Please either import all current layer with or without `self::` prefix. Mixing the two styles can cause confusion.
  --> tests/ui/import_self_style.fail.rs:12:9
   |
12 |         self::MyDepInThisLayer,
   |         ^^^^
